controllers/dataprotection/backup_controller.go

```diff
@@ -413,6 +412,16 @@ func (r *BackupReconciler) prepareBackupRequest(
 					backupSchedule.Namespace, backupSchedule.Name)
 			}
 		}
+
+		// validate parameters
+		withParameters := []string{}
+		if actionSet.Spec.Backup != nil {
+			withParameters = actionSet.Spec.Backup.WithParameters
+
+		}
+		if err := dputils.ValidateParameters(actionSet.Spec.ParametersSchema, withParameters, backup.Spec.Parameters); err != nil {
```

Comment:
it is better to validate alls conditions in a function. and you can input a actionSet object and a bool to check if ti is backup.